tests/op_coreutils/bigquery/test_write.py

```diff
@@ -0,0 +1,260 @@
+# -*- coding: utf-8 -*-
+# test_write.py
+
+from unittest.mock import patch, MagicMock
+import polars as pl
+import pytest
+import numpy as np
+import pandas as pd
+
+from op_coreutils.bigquery.write import (
+    init_client,
+    ensure_valid_dt,
+    overwrite_table,
+    overwrite_partition,
+    overwrite_partitions,
+    OPLabsBigQueryError,
+    OPLabsEnvironment,
+    upsert_partition,
+)
+
+
+# Generate a larger sample DataFrame for testing
+date_range = pd.date_range("2024-01-01", periods=100, freq="D")
+num_rows = 1000
+
+# Create random data for each column
+test_df = pl.DataFrame(
+    {
+        "dt": np.random.choice(date_range, num_rows),
+        "data": np.random.randint(100, 500, size=num_rows),
+        "category": np.random.choice(["A", "B", "C", "D"], num_rows),
+        "value": np.random.uniform(1.0, 10.0, size=num_rows).round(2),
+    }
+)
+
+# Sort by 'dt' for consistency
+test_df = test_df.sort("dt")
+
+
+def reset_client():
+    """Utility function to reset the global _CLIENT variable."""
+    import op_coreutils.bigquery.write
+
+    op_coreutils.bigquery.write._CLIENT = None
+
+
+@patch("op_coreutils.bigquery.write.get_credentials")
+@patch("op_coreutils.bigquery.write.bigquery.Client")
+def test_init_client(mock_bigquery_client, mock_get_credentials):
```

Comment:
I'm not sure testing the initialization method is warranted. It seems that for the most part, the test is checking that the mocks where set up correctly. Maybe consider removing this test?